### PR TITLE
Add close function

### LINKED ISSUE #214: Provide method to kill bridge
Currently, it is impossible to kill the bridge via publicly accessible methods. While the appservice part can be killed via `bridge.appService.close()`, we still have the intent culling loop at https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/src/bridge.js#L413 , which has to be manually killed with `clearTimeout(bridge._intentLastAccessedTimeout)`.

The bridge should instead expose a `killBridge` function that performs both of these, and whatever is needed in the future to ensure the bridge terminates.

This causes an issue with integration tests because the tests would never end with a dangling loop.

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #214

### CODE REVIEW

File: src/bridge.ts [hunk 1]
```diff
@@ -1212,6 +1212,19 @@ export class Bridge {
...
+     * @returns Resolves when the appservice HTTP listener has stopped
+     */
+    public async close() {
+        if (this.intentLastAccessedTimeout) {
+            clearTimeout(this.intentLastAccessedTimeout);
+        }
+        if (this.appservice) {
+            await this.appservice.close();
+        }
+    }
```
Comment (jaller94) [inline src/bridge.ts:1226]: I think we should reset these or have a boolean flag that prevents the Bridge from being reused.
```suggestion
        if (this.intentLastAccessedTimeout) {
            clearTimeout(this.intentLastAccessedTimeout);
            this.intentLastAccessedTimeout = null;
        }
        if (this.appservice) {
            await this.appservice.close();
            this.appservice = null;
        }
    }
```

**Review by jaller94 — APPROVED** [review]
Generally ok, but I would wish we planned for the object being reusable or prevent it from working again.

File: src/bridge.ts (hunk 1 above)
Comment (Half-Shot) [reply src/bridge.ts:1226]: Yeah, works for me.

**Comment by jaller94** [comment]
ops. I meant undefined for appservice.

File: src/bridge.ts [hunk 2]
```diff
@@ -1212,6 +1212,21 @@ export class Bridge {
...
+     * @returns Resolves when the appservice HTTP listener has stopped
+     */
+    public async close() {
+        if (this.intentLastAccessedTimeout) {
+            clearTimeout(this.intentLastAccessedTimeout);
+            this.intentLastAccessedTimeout = null;
+        }
+        if (this.appservice) {
+            await this.appservice.close();
+            this.appservice = null;
```
Comment (jaller94) [inline src/bridge.ts:1226]:
```suggestion
            this.appservice = undefined;
```

**Pull request merged**